firebase-perf/src/main/java/com/google/firebase/perf/injection/modules/GaugeCollectorsModule.java

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.injection.modules;
+
+import static android.system.Os.sysconf;
+
+import android.os.Build;
+import android.system.OsConstants;
+import dagger.Module;
+import dagger.Provides;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Provider for {@link com.google.firebase.perf.gauges}.
+ *
+ * @hide
+ */
+@Module
+public class GaugeCollectorsModule {
```

Comment:
Is it preferred to create module for specific classes like `Cpu/MemoryGaugeCollector` in this case or should it be functionality wise for e.g. `GaugeModule` for all gauge related things?

The reason I ask this is because I will have some stuff for `GaugeMetadataManager` and I think it would make more sense to have it unified functionality wise.

Reply:
Yes. I created the module because I think `Cpu/MemoryGaugeCollector` will be a self-contained module. We can also group them by directory (although some directories need to be restructured)

Reply:
+1 on directory restructure but cases where functionality is well contained lets have a single module. Can you rename this module to `GaugeModule` in that case? 

fyi I have created `GaugeModule` class for `GaugeMetadaManager` but whosoever push the changes last can resolve any merge errors.

Reply:
Sounds good!